sympy/core/power.py

```diff
@@ -861,9 +861,10 @@ def _check(ct1, ct2, old):
                         as_int(pow, strict=False)
                         combines = True
                     except ValueError:
-                        combines = isinstance(Pow._eval_power(
-                            Pow(*old.as_base_exp(), evaluate=False),
-                            pow), (Pow, exp, Symbol))
+                        b, e = old.as_base_exp()
+                        # These conditions ensure that (b**e)**f == b**(e*f) for any f
+                        combines = b.is_positive and e.is_real or b.is_nonnegative and e.is_nonnegative
```

Comment:
```suggestion
                        obase, oexp =old.as_base_exp()
                        p = Pow._eval_power(
                            Pow(obase, oexp, evaluate=False), pow)
                        if p is None:
                            combines = False
                        else:
                            pbase = p.as_base_exp()[0]
                            combines = pbase == obase

```

Reply:
Note that this will now keep `sqrt(x).subs(x**2,y)` from changing but will allow `sqrt(p).subs(p**2,y)` to become `sqrt(sqrt(p))` if `p` is positive. So to disable algebraic subs, Integral need only use vanilla symbols.